src/mptool.f90

```diff
@@ -132,7 +152,134 @@ real(mytype) function  pmax_mytype(var)
                                                     mpi_comm_world,ierr)
     !
   end function pmax_mytype
+
+  real(mytype) function  pmax_mytype_array(var)
+    !
+    ! arguments
+    real(mytype),intent(in) :: var(:)
+    !
+    ! local data
+    integer :: ierr
+    !
+    call mpi_allreduce(var,pmax_mytype_array,size(var),real_type,mpi_max,    &
+                                                    mpi_comm_world,ierr)
+    !
+  end function pmax_mytype_array
+
+  real(mytype) function  pmin_mytype_array(var)
```

Comment:
Please make the size of the output explicit so that it is easier to read / understand the function

Reply:
Sometimes, the size of the array is not explicit. 
I have added an optional parameter to handle it

Reply:
Without the optional parameter was also possible using `result(output)` and real(mytype) :: output(size(var))

Reply:
I don't think the function as written can work, the array being reduced into is never allocated, consider the simplified version below with the Cray compiler:
```
======= func_arr.f90
program test

   real, dimension(5) :: b

   print *, pmin_mytype_array(b)

contains

real function pmin_mytype_array(var)

    real, intent(in) :: var(:)
    real, dimension(size(var)) :: a

    a = 7
    pmin_mytype_array = a

end function

end program
======= end func_arr.f90

> ftn func_arr.f90 

    pmin_mytype_array = a
                      ^   
ftn-324 ftn: ERROR PMIN_MYTYPE_ARRAY, File = func_arr.f90, Line = 15, Column = 23 
  Assignment of a rank 1 expression to a rank 0 variable is not allowed.

Cray Fortran : Version 15.0.0 (20221026200610_324a8e7de6a18594c06a0ee5d8c0eda2109c6ac6)
Cray Fortran : Compile time:  0.0044 seconds
Cray Fortran : 19 source lines
Cray Fortran : 1 errors, 0 warnings, 0 other messages, 0 ansi
Cray Fortran : "explain ftn-message number" gives more information about each message.
```
so I think the parameter cannot be optional, mathrack's solution should solve this.

Reply:
followed mathrack's solution. I missunderstood the point...